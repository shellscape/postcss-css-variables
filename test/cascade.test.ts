import { fixture, process } from './helpers';

const tests = {
  'cascade to nested multiple rules': 'multiple-on-nested-rules',
  'cascade to nested rules': 'nested-rules',
  'cascade to nested rules in the proper scope': 'nested-rules-in-proper-scope',
  'cascade with calc-expression to nested rules': 'calc-expression-on-nested-rules'
};

describe('cascading', async () => {
  test.each(Object.entries(tests))('%s', async (_desc, file) => {
    const css = await fixture(`cascade/${file}`);
    const result = await process(css);
    expect(result).toMatchSnapshot();
  });
});
