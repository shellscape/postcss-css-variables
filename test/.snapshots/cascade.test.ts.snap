// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`cascading > cascade to nested multiple rules 1`] = `".a{width:150px;.simple{color:#FFF}}.b{width:150px;.simple{color:#FFF}}.a{width:150px;a{color:#FFF}label{color:#FFF}&:after{color:#FFF}}.a{width:150px;a{background:#FFF;ol{width:150px}ul{width:150px}}label{background:#FFF;ol{width:150px}ul{width:150px}}}.b{width:150px;a{background:#FFF;ol{width:150px}ul{width:150px}}label{background:#FFF;ol{width:150px}ul{width:150px}}}"`;

exports[`cascading > cascade to nested rules 1`] = `".box-foo{width:150px;.box-bar{width:150px}@media (max-width: 800px){.box-bar{width:300px}}}@media (max-width: 800px){.box-foo{width:300px}}"`;

exports[`cascading > cascade to nested rules in the proper scope 1`] = `".box-foo{width:333px;.box-bar{width:333px}}.box-baz{width:200px}@media (max-width: 800px){.box-baz{width:550px}}"`;

exports[`cascading > cascade with calc-expression to nested rules 1`] = `".box-foo{width:calc(50px + 150px);.box-bar{width:calc(50px + 150px)}@media (max-width: 800px){.box-bar{width:calc(50px + 300px)}}}@media (max-width: 800px){.box-foo{width:calc(50px + 300px)}}"`;
